python-package/packager/nativelib.py

```diff
@@ -0,0 +1,158 @@
+"""
+Functions for building libxgboost
+"""
+import logging
+import os
+import pathlib
+import shutil
+import subprocess
+import sys
+from platform import system
+from typing import Optional
+
+from .build_config import BuildConfiguration
+
+
+def _lib_name() -> str:
+    """Return platform dependent shared object name."""
+    if system() in ["Linux", "OS400"] or system().upper().endswith("BSD"):
+        name = "libxgboost.so"
+    elif system() == "Darwin":
+        name = "libxgboost.dylib"
+    elif system() == "Windows":
+        name = "xgboost.dll"
+    else:
+        raise NotImplementedError(f"System {system()} not supported")
+    return name
+
+
+def build_libxgboost(
+    cpp_src_dir: pathlib.Path,
+    *,
+    build_dir: pathlib.Path,
+    build_config: BuildConfiguration,
```

Comment:
I personally like using keyword-only arguments, because I am being explicitly about which arguments correspond to which argument. Do you feel that this adds complexity?

Reply:
I don't mind keyword-only arguments, but I find them odd when they come without defaults. More importantly, in this case I'm not sure why you would have one positional argument and then have two keyword-only arguments with no defaults. That's the main thing that makes for a confusing API IMO. I would make them all keyword-only if that's your preference.

Reply:
I removed all `*` from function signatures.